### PR TITLE
toplevel displays help if no program matches command arguments

### PR DESCRIPTION
**Pull request by BebeSparkelSparkel**
I found it frustrating that if a command was entered incorrectly nothing was displayed and it seemed like the program was executed correctly.

`toplevel` now displays the documentation and exits with failure code if no program is matched.

### CODE REVIEW

File: src/Options/Commander.hs [hunk 1]
```diff
@@ -528,8 +529,17 @@ flag = FlagProgramT
...
+         -> ProgramT (Named s & ("help" & Raw + p + Raw)) m ()
+-- toplevel p = named (sub (usage @(Named s & ("help" & Raw + p))) <+> p)
+toplevel p = named $ helpSubprogram <+> p <+> helpIfMissed
+  where
+  helpSubprogram = sub $ usage @(Named s & ("help" & Raw + p))
+  helpIfMissed :: ProgramT Raw m ()
+  helpIfMissed = raw $ liftIO do
+    putStrLn "usage:"
+    putStrLn (document @p)
+    exitFailure
```
Comment (SamuelSchlesinger) [inline src/Options/Commander.hs:541]: Yeah, I totally agree with this. I think it means that the example can be simplified as well, please include that in this PR or explain why that is not the case, as a demonstration of the use of this function.

**Review by SamuelSchlesinger — CHANGES_REQUESTED** [review]
Awesome! For some reason I thought I tried this and it failed to work.

**Comment by BebeSparkelSparkel** [comment]
being discussed in #42 and what should be included

File: src/Options/Commander.hs (hunk 1 above)
Comment (SamuelSchlesinger) [reply src/Options/Commander.hs:541]: Bump!

Comment (BebeSparkelSparkel) [reply src/Options/Commander.hs:541]: could you comment on #42

**Pull request closed without merging**